lib/go-qmstr/database/diagnosticNode.go

```diff
@@ -13,17 +13,19 @@ import (
 )
 
 // AddDiagnosticNodes stores the given DiagnosticNodes in a PackageNode or FileNode identified by the nodeID
-func (db *DataBase) AddDiagnosticNodes(nodeID string, diagnosticnodes ...*service.DiagnosticNode) error {
+func (db *DataBase) AddDiagnosticNodes(nodeID string, diagnosticnode *service.DiagnosticNode) error {
```

Comment:
You changed the semantics to process only one diagnostic node, but the function is still named AddDiagnosticNode**s**